besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2715,12 +2735,14 @@ private String generateConfigurationOverview() {
       builder.setHighSpecEnabled();
     }
 
-    if (DataStorageFormat.BONSAI.equals(getDataStorageConfiguration().getDataStorageFormat())
-        && getDataStorageConfiguration().getBonsaiLimitTrieLogsEnabled()) {
-      builder.setLimitTrieLogsEnabled();
-      builder.setTrieLogRetentionLimit(getDataStorageConfiguration().getBonsaiMaxLayersToLoad());
-      builder.setTrieLogsPruningWindowSize(
-          getDataStorageConfiguration().getBonsaiTrieLogPruningWindowSize());
+    if (DataStorageFormat.BONSAI.equals(getDataStorageConfiguration().getDataStorageFormat())) {
+      final DiffBasedSubStorageConfiguration subStorageConfiguration =
+          getDataStorageConfiguration().getDiffBasedSubStorageConfiguration();
+      if (subStorageConfiguration.getLimitTrieLogsEnabled()) {
+        builder.setLimitTrieLogsEnabled();
+        builder.setTrieLogRetentionLimit(subStorageConfiguration.getMaxLayersToLoad());
+        builder.setTrieLogsPruningWindowSize(subStorageConfiguration.getTrieLogPruningWindowSize());
+      }
     }
```

Comment:
nit: just a suggestion - as you're only modifying the builder you could create a method called `ConfigurationOverviewBuilder.accept(DataStorageConfiguration)` where you move in all of this logic and maybe make setters private so that no one can do `setLimitTrieLogsEnabled` if not for a BONSAI storage.

Reply:
The logic for modifying certain things only when in Bonsai mode is present in several places within BesuCommand. I noticed this while making my changes. I believe there should be another PR to possibly revisit the logic in BesuCommand at this level in order to avoid having `DataStorageFormat.BONSAI.equals(dataStorageConfiguration.getDataStorageFormat()` . I can try to create another PR to address this later, but I don't want to increase the size of this one